### PR TITLE
Added urlquery middleware, resolves #174.

### LINKED ISSUE #174: Suggest middleware for request URL.Query()
I have several middlewares that need query parameters.  However, each call to r.URL.Query() parses the query string, and this is not a no-op.

Therefore I have made a middleware which does this once, and stores it in c.Env.

Would you accept a pull request for this?

### PR DESCRIPTION
**Pull request by tesujimath**

### CODE REVIEW

File: web/middleware/urlquery.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package middleware
+
+import (
+	"github.com/zenazn/goji/web"
+	"net/http"
+)
+
+// UrlQueryKey is the context key for the URL Query
+const UrlQueryKey string = "urlquery"
```
Comment (zenazn) [inline web/middleware/urlquery.go]: URL should be capitalized here and elsewhere (e.g., https://golang.org/pkg/net/http/#ProxyURL)

**Comment by zenazn** [comment]
Thanks for this!

**Comment by tesujimath** [comment]
Any idea if/when you are likely to merge this?

**Comment by zenazn** [comment]
Ah, sorry—GitHub didn't notify me when you made that update (0835e6a)

**Pull request merged**